pkg/deploy/lattice/iamauthpolicy_manager.go

```diff
@@ -10,39 +10,127 @@ import (
 )
 
 type IAMAuthPolicyManager struct {
-	Cloud pkg_aws.Cloud
+	cloud pkg_aws.Cloud
+}
+
+func NewIAMAuthPolicyManager(cloud pkg_aws.Cloud) *IAMAuthPolicyManager {
+	return &IAMAuthPolicyManager{cloud: cloud}
 }
 
 func (m *IAMAuthPolicyManager) Put(ctx context.Context, policy model.IAMAuthPolicy) (model.IAMAuthPolicyStatus, error) {
+	switch policy.Type {
+	case model.ServiceNetworkType:
+		return m.putSn(ctx, policy)
+	case model.ServiceType:
+		return m.putSvc(ctx, policy)
+	default:
+		panic("unknown policy resource type: " + policy.Type)
+	}
+}
+
+func (m *IAMAuthPolicyManager) putSn(ctx context.Context, policy model.IAMAuthPolicy) (model.IAMAuthPolicyStatus, error) {
+	sn, err := m.cloud.Lattice().FindServiceNetwork(ctx, policy.Name, "")
+	if err != nil {
+		return model.IAMAuthPolicyStatus{}, err
+	}
+	resourceId := *sn.SvcNetwork.Id
+	err = m.enableSnIAMAuth(ctx, resourceId)
+	if err != nil {
+		return model.IAMAuthPolicyStatus{}, err
+	}
+	err = m.putPolicy(ctx, resourceId, policy.Policy)
+	if err != nil {
+		return model.IAMAuthPolicyStatus{}, err
+	}
+	return model.IAMAuthPolicyStatus{ResourceId: resourceId}, nil
+}
+
+func (m *IAMAuthPolicyManager) putSvc(ctx context.Context, policy model.IAMAuthPolicy) (model.IAMAuthPolicyStatus, error) {
+	svc, err := m.cloud.Lattice().FindService(ctx, policy.Name)
+	if err != nil {
+		return model.IAMAuthPolicyStatus{}, err
+	}
+	resourceId := *svc.Id
+	err = m.enableSvcIAMAuth(ctx, resourceId)
```

Comment:
If I understand it correctly from our previous standup meeting, for this issue: https://github.com/aws/aws-application-networking-k8s/issues/467

When create the AuthPolicy, the order should be: (not correct in current PR)
1. PutAuthPolicy
2.  Change AuthType to AWS_IAM

When create the AuthPolicy, the order should be: (correct in current PR)
1. Change AuthType to NONE
2. DeleteAuthPolicy

Do you need to change creation handling order?

Reply:
to put policy we first enable iam and then put policy, for delete we disable iam and remove policy

Reply:
@mohakkohli Could you help to double confirm?